### PR TITLE
Fix for len intrinsic returning int*8 in some cases

### PR DESCRIPTION
**Pull request by kiranchandramohan**
Flang's implementation of len intrinsic returns the length attribute.
After large array support was introduced, this attribute is stored in
an integer of kind=8. The standard stipulates that the integer
returned should be of the default (kind=4). Fix is to convert to the
right kind.

### CODE REVIEW

**Review by gklimowicz — APPROVED** [review]
This passes our tests on OpenPOWER for LLVM 9 and LLVM 10.

File: test/f90_correct/inc/fix_len_returning_i8.mk [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#
+# Copyright (c) 2019, Arm Ltd.  All rights reserved.
```
Comment (kiranchandramohan) [inline test/f90_correct/inc/fix_len_returning_i8.mk:2]: Change/Remove copyright before submission.

**Review by shivaramaarao — APPROVED** [review]
LGTM

File: test/f90_correct/len_intrinsic.f90 [hunk 2]
```diff
@@ -0,0 +1,47 @@
+!** Copyright (c) 2019, Arm Ltd.  All rights reserved.
```
Comment (bryanpkc) [inline test/f90_correct/len_intrinsic.f90:1]: This file should be moved to test/llvm_ir_correct/. Use the same license notice as in other files.

File: test/f90_correct/len_intrinsic.f90 [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+! RUN: %flang -S -emit-llvm %s -o - | FileCheck %s -check-prefix=NO_CONVERT
+! RUN: %flang -fdefault-integer-8 -S -emit-llvm %s -o - | FileCheck %s -check-prefix=CONVERT_I8
+! RUN: %flang -fdefault-integer-8 -S -emit-llvm %s -o - | FileCheck %s -check-prefix=NO_CONVERT_I8
+subroutine s1(x)
+  character(len=x) :: s1_str
+  integer :: s1_test_len
+  s1_test_len = len(s1_str)
+  print *, s1_test_len
+end subroutine
+! //CONVERT: define void @s1_
```
Comment (bryanpkc) [inline test/f90_correct/len_intrinsic.f90:27]: The `//` should be deleted. The correct FileCheck comment prefix should be `! CONVERT:`.

Comment (kiranchandramohan) [reply test/f90_correct/len_intrinsic.f90]: done,

File: test/f90_correct/len_intrinsic.f90 (hunk 2 above)
Comment (kiranchandramohan) [reply test/f90_correct/len_intrinsic.f90]: done.

File: test/f90_correct/inc/fix_len_returning_i8.mk (hunk 1 above)
Comment (kiranchandramohan) [reply test/f90_correct/inc/fix_len_returning_i8.mk]: done.

**Review by bryanpkc — APPROVED** [review]
Thanks, LGTM now.

**Pull request merged**